plugins/woocommerce/includes/class-wc-form-handler.php

```diff
@@ -916,7 +916,7 @@ private static function add_to_cart_handler_variable( $product_id ) {
 		}
 
 		if ( false !== WC()->cart->add_to_cart( $product_id, $quantity, $variation_id, $variations ) ) {
-			wc_add_to_cart_message( array( $product_id => $quantity ), true );
+			wc_add_to_cart_message( array( $product_id => $quantity ), true, false, $variation_id );
```

Comment:
Given that we support sending multiple product arrays to `wc_add_to_cart_message` method, adding another param that just accepts a single `$variation_id` won't support the use case of sending multiple variation_ids to this method in one go. It might be better to send in `$variation_id` instead of `$product_id` when it makes sense, we can work it out in the method later.

```suggestion
			$product_or_variation_id = ! empty( $variation_id ) ? $variation_id : $product_id;
			wc_add_to_cart_message( array( $product_or_variation_id => $quantity ), true, false, $variation_id );
```